dbt_platform_helper/providers/cache.py

```diff
@@ -0,0 +1,74 @@
+import os
+from datetime import datetime
+from pathlib import Path
+
+import yaml
+
+from dbt_platform_helper.constants import PLATFORM_HELPER_CACHE_FILE
+
+
+class CacheProvider:
+
+    def read_supported_versions_from_cache(self, resource_name):
+
+        platform_helper_config = self.__read_file_as_yaml(PLATFORM_HELPER_CACHE_FILE)
+
+        return platform_helper_config.get(resource_name).get("versions")
+
+    def write_to_cache(self, resource_name, supported_versions):
+
+        platform_helper_config = {}
+
+        if os.path.exists(PLATFORM_HELPER_CACHE_FILE):
+            platform_helper_config = self.__read_file_as_yaml(PLATFORM_HELPER_CACHE_FILE)
+
+        cache_dict = {
+            resource_name: {
+                "versions": supported_versions,
+                "date-retrieved": datetime.now().strftime("%d-%m-%y %H:%M:%S"),
+            }
+        }
+
+        platform_helper_config.update(cache_dict)
+
+        with open(PLATFORM_HELPER_CACHE_FILE, "w") as file:
+            file.write("# [!] This file is autogenerated via the platform-helper. Do not edit.\n")
+            yaml.dump(platform_helper_config, file)
+
+    def cache_refresh_required(self, resource_name) -> bool:
+        """
+        Checks if the platform-helper should reach out to AWS to 'refresh' its
+        cached values.
+
+        An API call is needed if any of the following conditions are met:
+            1. No cache file (.platform-helper-config.yml) exists.
+            2. The resource name (e.g. redis, opensearch) does not exist within the cache file.
+            3. The date-retrieved value of the cached data is > than a time interval. In this case 1 day.
+        """
+
+        if not self.__platform_helper_cache_exists():
+            return True
+
+        platform_helper_config = self.__read_file_as_yaml(PLATFORM_HELPER_CACHE_FILE)
+
+        if platform_helper_config.get(resource_name):
+            return self.__check_if_cached_datetime_is_greater_than_interval(
+                platform_helper_config[resource_name].get("date-retrieved"), 1
+            )
+
+        return True
+
+    def __check_if_cached_datetime_is_greater_than_interval(self, date_retrieved, interval_in_days):
+
+        current_datetime = datetime.now()
+        cached_datetime = datetime.strptime(date_retrieved, "%d-%m-%y %H:%M:%S")
+        delta = current_datetime - cached_datetime
+
+        return delta.days > interval_in_days
+
+    def __read_file_as_yaml(self, file_name):
+
+        return yaml.safe_load(Path(file_name).read_text())
+
+    def __platform_helper_cache_exists(self):
+        return os.path.exists(PLATFORM_HELPER_CACHE_FILE)
```

Comment:
```suggestion
        return os.path.exists(self.cache_file)
```

Actually, you might as well inline this. The method is only used once and doesn't add anything to the readability of the code.

Reply:
See discussion with Kate - the alternative here is to use fakefs in the pytests. which has introduced some flakeyness before. 

Also I forgot to add a method call for this under write_to_cache()

Reply:
If it's used twice that's fine.

Re the tests, this would be a legitimate use case for patching. You could patch `os.path.exists` inside the test.